src/ast/semantic_analyser.cpp

```diff
@@ -1499,6 +1499,32 @@ void SemanticAnalyser::visit(Binop &binop)
           << " operator can not be used on expressions of types " << lhs << ", "
           << rhs;
     }
+    else if (binop.op == Parser::token::EQ && !binop.left->is_literal &&
+             binop.right->is_literal)
+    {
+      auto llen = binop.left->type.GetNumElements();
+      auto rlen = bpftrace_.get_string_literal(binop.right).size();
+      if (rlen > llen)
+      {
+        LOG(ERROR, binop.left->loc + binop.loc + binop.right->loc, err_)
```

Comment:
Dont think we have to keep two cases here. Something simple like this should be enough:

```
Literal is longer than the variable string, condition will always be false
```